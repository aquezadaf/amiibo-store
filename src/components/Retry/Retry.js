import React from "react";
import { useTranslation } from "react-i18next";
import PropTypes from "prop-types";
import styles from "./Retry.module.css";

function Retry(props) {
  const { onPress } = props;
  const { t } = useTranslation();
  return (
    <div className={styles.container}>
      <div className={styles.text}>{t("genericError")}</div>
      <button className={styles.button} onClick={onPress}>
        {t("retry")}
      </button>
    </div>
  );
}

Retry.propTypes = {
  onPress: PropTypes.func.isRequired,
};

export default Retry;
