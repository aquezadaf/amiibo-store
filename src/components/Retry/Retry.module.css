.text {
    font-size: 15px;
    font-weight: bold;
    margin-bottom: 10px;
}

.button {
    width: 152px;
    font-size: 15px;
    color: #fff;
    border: 2px solid #fff;
    background-color: #0382b4;
    text-align: center;
    text-transform: uppercase;
    font-weight: 700;
    padding: 8px 0;
}
