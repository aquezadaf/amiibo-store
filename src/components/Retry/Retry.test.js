import React from "react";
import { shallow } from "enzyme";
import Retry from "./Retry";

it("renders retry", () => {
  const wrapper = shallow(<Retry onPress={() => {}} />);
  expect(wrapper.html()).toMatchSnapshot();
});

it("calls on press when clicking the button", () => {
  const pressMock = jest.fn();
  const wrapper = shallow(<Retry onPress={pressMock} />);
  wrapper.find("button").simulate("click");
  expect(pressMock).toHaveBeenCalled();
});
