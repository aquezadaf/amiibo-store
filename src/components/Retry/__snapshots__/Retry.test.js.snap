// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders retry 1`] = `"<div class=\\"container\\"><div class=\\"text\\">genericError</div><button class=\\"button\\">retry</button></div>"`;
